sdk/src/message.rs

```diff
@@ -165,9 +165,17 @@ pub struct Message {
 
 impl Sanitize for Message {
     fn sanitize(&self) -> std::result::Result<(), SanitizeError> {
-        if self.header.num_required_signatures as usize > self.account_keys.len() {
+        if self.header.num_required_signatures as usize
+            + self.header.num_readonly_unsigned_accounts as usize
+            > self.account_keys.len()
+        {
+            return Err(SanitizeError::IndexOutOfBounds);
+        }
+
+        if self.header.num_readonly_signed_accounts > self.header.num_required_signatures {
```

Comment:
I think this needs to be `>=`
If `num_readonly_signed_accounts == num_required_signatures`, there will be no signer that can be debited for the tx fees.

Reply:
That said, I assume we're including these just to be comprehensive about our transaction assumptions in one place. Because the sigverify check always happens first, right? If so, transactions can't make it to `sanitize` in `Bank::check_refs()` and still be in violation of this check.

Reply:
yea, sigverify check should filter this in the full validator. As you said, to be comprehensive.

Reply:
Will change to <=

Reply:
We do have this though:
https://github.com/solana-labs/solana/blob/79a6b4b596b522f321c70ad26b6b3b3c1f7ccf25/runtime/src/bank.rs#L4589

Reply:
Hmm. it allows a read-only account to pay the fee, but then does not collect that account to store... tricky..

Reply:
Oh that is tricky. Didn't realize we supported that, and it seems like the sigverify check actually prevents successful zero-signature transactions from being processed, even though the bank allows it.

Do we want to support zero-signature transactions?

If not, we can just ax this test.

Reply:
I adjusted the test to expect an error.